lib/realtime.js

```diff
@@ -272,16 +272,24 @@ function isReady () {
 }
 
 function extractNoteIdFromSocket (socket) {
-  if (!socket || !socket.handshake || !socket.handshake.headers) {
+  if (!socket || !socket.handshake) {
     return false
   }
-  var referer = socket.handshake.headers.referer
-  if (!referer) {
+  if (socket.handshake.query) {
```

Comment:
We should evaluate the entire path, otherwise it'll may cause breaks.

`socket.handshake.query && socket.handshake.query.noteId`

Reply:
this is implicitly done by the previous return (line 275 and 276). IF we end up here, we _know_ that sockt and socket handshake must exist.

Reply:
On could argue that this breaks at some point down the road if this code gets modified by unsuspecting people. But that's what regression tests are for, IMO.

Reply:
@SISheogorath You are right, it actually should break (due to query parameters like `transport`).

This time I actually tested the update, and the reconnect worked without reloading the page.